src/net.cpp

```diff
@@ -2331,6 +2332,11 @@ bool CAddrDB::Read(CAddrMan& addr)
 
     CDataStream ssPeers(vchData, SER_DISK, CLIENT_VERSION);
 
+    return Read(addr, ssPeers, hashIn);
+}
+
+bool CAddrDB::Read(CAddrMan& addr, CDataStream& ssPeers, uint256& hashIn)
+{
     // verify stored checksum matches input data
     uint256 hashTmp = Hash(ssPeers.begin(), ssPeers.end());
```

Comment:
Could make sense to move the hash check to the outer read function - after all, it's not part of the inner stream but of the wrapping file.
Would simplify testing as there'd be no need to pass hashIn here.